kurobako_core/src/problem.rs

```diff
@@ -76,8 +86,23 @@ impl ProblemSpecBuilder {
             params_domain,
             values_domain,
             steps,
+            reference_point: self.reference_point,
         })
     }
+
+    fn validate(&self) -> Result<()> {
+        if let Some(reference_point) = &self.reference_point {
+            if reference_point.len() != self.values.len() {
+                track_panic!(
+                    ErrorKind::InvalidInput,
+                    "Unexpected dimensions of values {} and reference point {}",
+                    self.values.len(),
+                    reference_point.len()
+                )
+            }
```

Comment:
We can use `track_assert_eq!` as follows:
```suggestion
            track_assert_eq!(
                reference_point.len(),
                self.values.len(),
                ErrorKind::InvalidInput,
                "Unexpected dimensions of values {} and reference point {}",
                self.values.len(),
                reference_point.len()
            )
```